src/util/dynamicConfigParser.ts

```
@@ -6,6 +6,54 @@
+    return this.useHasDynamicConfigFlag && destination?.hasDynamicConfig === false;
+  }
+
+  /**
+   * Utility function to check if events should be grouped by destination config
+   * based on the USE_HAS_DYNAMIC_CONFIG_FLAG environment variable and the hasDynamicConfig flag.
+   *
+   * @param destination - The destination object containing the hasDynamicConfig flag
+   * @returns true if events should be grouped by destination config, false otherwise
+   */
+  public static shouldGroupByDestinationConfig(destination: any): boolean {
+    if (!this.useHasDynamicConfigFlag) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)